### ISSUE TITLE
NameFieldResolver class is not public

### ISSUE THREAD

**Issue by Shane32** [graphql-dotnet/graphql-dotnet#1378]
I'm trying to create a custom IExecutionStrategy, but the default implementation contains the following line of code within the protected virtual ExecuteNodeAsync method, which cannot run in a separate library due to NameFieldResolver being an internal class:
``` C#
var resolver = node.FieldDefinition.Resolver ?? new NameFieldResolver();
```
I suggest that NameFieldResolver be a public class.  This allows for custom implementations of IExecutionStrategy to maintain 100% compatibility with other code without rewriting/copying substantial amounts of graphql code.  I can submit a pull request if desired.

**Comment by sungam3r** [comment]
Ok.

**Comment by sungam3r** [comment]
Closed by #1379
